src/main/java/ch/njol/skript/util/slot/EquipmentSlot.java

```diff
@@ -180,9 +181,18 @@ public ItemStack getItem() {
 	
 	@Override
 	public void setItem(final @Nullable ItemStack item) {
-		slot.set(e, item);
-		if (e.getHolder() instanceof Player)
-			PlayerUtils.updateInventory((Player) e.getHolder());
+		// Small bandaid for Spigot's AssertionError "TRAP" (crashing the server) when setting a player's tool to air/null during some events
+		// We add a small delay, to set the item to air/null after Minecraft has handled changing the item amount
+		if (e.getHolder() instanceof Player && (slot == EquipSlot.TOOL || slot == EquipSlot.OFF_HAND) && (item == null || item.getType() == Material.AIR)) {
+			Bukkit.getScheduler().scheduleSyncDelayedTask(Skript.getInstance(), () -> {
```

Comment:
It might not be a good idea to do this each time a slot is set, as far I've read the issue mainly happens on interaction events so we should probably handle it there

Reply:
You're probably right.
Doing some testing this only seems happen in a right click event.
It doesn't appear to happen in a place event, break event or left click event.

I have an idea I will try, and if it works ill update this PR and report back.